### PR TITLE
Emphasize relative importance and harden + consolidate "Effects of Priority Hints"

### LINKED ISSUE #32: Spec should specify (and emphasize) "relative", importance
At the Web Perf WG call today, it seemed that one of @toddreifsteck's more prominent hesitations revolved around the notion that Priority Hints might be trying to shift priorities among the many and different types (destinations, really) of resources. For example, a concern would have been "Well would a high importance image get prioritized higher than a low importance CSS file? And what if it did in Chrome but not Edge? etc." This sounded a bit like the original intent of Priority Hints (I recall group names and https://github.com/WICG/priority-hints/issues/7#issuecomment-331070080, which seems solved by specifying and emphasizing _relative_ importance). I think a breakthrough we had in the conversation was when we mentioned that the prioritization of resources should be relative between types/destinations. For example, no matter what prioritization scheme might be used in Edge vs Chrome vs FF vs Safari, low-importance images should be prioritized lower than *other non-low-importance* images. The same goes for scripts, fetch()es, etc. Modifying the priority of a request relative to other requests of the same type definitely seems like something we'd like to make clear(er).

Assigning this to Addy right now as he's been digging around the spec quite a bit lately, but happy to pick this up if needbe after some more implementation work.

----

(aside from the main point of this issue:)

Another very reasonable hesitation we got revolved around the fact that it might be helpful to fully specify exactly how a resource gets prioritized, and some of the surrounding infrastructure, values, and logic so we could have this work dependably between the different implementations. While this is very valuable, I (and I believe @yoavweiss) think that Priority Hints would actually be a good step to getting us there, and is not a blocker so much as a catalyst to that conversation (which there has been some, albeit not tons, of talk around). For example, by talk about how `importance=high` async scripts should be prioritized higher than `importance=low` async scripts, we're already making relative comparisons that the Fetch Standard does not yet talk about.

### PR DESCRIPTION
**Pull request by domfarolino**
This PR attempts to fix #32. A few things it does:

 - Specifies exactly what the expected effects of Priority Hints are in a new section, since it is possible to see the current state of this as vague and under-specified.
 - Includes an example showing that Priority Hints are meant to be a relative influencer, ideally clearing up some confusion in https://github.com/mozilla/standards-positions/issues/25 and elsewhere.
- Attempts to consolidate references to what exactly the expected effects of the hints are. There are some cases where we reference things like "...affect the HTTP/2 weights and dependencies of a request" from several locations. I attempt to cut a few of these out in favor of referencing the newly added section. This might make the document more maintainable.
 - Adds more definition dependencies that I refer to in this PR + the HTML Standard biblio entry
 - Changes a few instances of "may" to "might", in an effort to use more non-normative language where need-be (please see https://infra.spec.whatwg.org/#conformance)

/cc @kenjibaheux @kinu @slightlyoff @toddreifsteck @wanderview

----

With some of the newer changes to the spec, the Explainer doc is getting more and more out of date. Have we looked into what we do with the explainer once the spec gets more mature? Remove it? Delete it?

### CODE REVIEW

**Comment by KenjiBaheux** [comment]
I've read the proposed changes and they seemed reasonable to me.
But I defer to Addy for the approval.

**Review by addyosmani — APPROVED** [review]
LGTM.

File: index.html
Comment (yoavweiss) [inline index.html]: destination is a property of the request, no?

File: index.html
Comment (yoavweiss) [inline index.html]: I think we can remove the "sketch" part

File: index.html
Comment (yoavweiss) [inline index.html]: s/compliment/complement/

File: index.html
Comment (yoavweiss) [inline index.html]: Maybe rephrase? e.g. "Since user agents' resource loading pipeline is largely unspecified, this document doesn't mandate specific mapping between priority hints and request prioritization. However, it does describe some ways in which..."

File: index.html
Comment (yoavweiss) [inline index.html]: s/browser/user agent/

**Review by yoavweiss — COMMENTED** [review]
Thanks for working on this! :) A few comments

**Comment by yoavweiss** [comment]
@toddreifsteck - can you take a look and say if this addresses your concerns?

**Comment by domfarolino** [comment]
Thanks for the review everyone! I've addressed Yoav's comments. It'd be great if Todd could take a look and share thoughts, but won't I block merging in the mean time.

**Pull request merged**
